core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -849,17 +846,55 @@ public Graph build() {
+            if (store.isSupported(AddNamedView.class)) {
+                final OperationHandler addNamedViewHandler = store.getOperationHandler(AddNamedView.class);
+                final String suffix = addNamedViewHandler.getClass().isAssignableFrom(AddNamedViewHandler.class) ? ((AddNamedViewHandler) addNamedViewHandler).getSuffixCacheName() : properties.getCacheServiceNamedViewSuffix(config.getGraphId());
+                if (!hasHook(hooks, NamedViewResolver.class)) {
+                    LOGGER.warn(HANDLER_WAS_SUPPLIED_BUT_WITHOUT_A_ADDING_WITH_SUFFIX, AddNamedViewHandler.class.getSimpleName(), NamedViewResolver.class.getSimpleName(), NamedViewResolver.class.getSimpleName(), suffix);
+                    hooks.add(0, new NamedViewResolver(suffix));
+                } else {
+                    final NamedViewResolver nvrHook = (NamedViewResolver) hooks.stream().filter(gh -> NamedViewResolver.class.isAssignableFrom(gh.getClass())).findAny().get();
+                    final String nvrSuffix = nvrHook.getSuffixCacheName();
+                    if (!suffix.equals(nvrSuffix)) {
+                        throw new GafferRuntimeException(String.format(HOOK_SUFFIX_ERROR_FORMAT_MESSAGE, NamedViewResolver.class.getSimpleName(), nvrSuffix, AddNamedViewHandler.class.getSimpleName(), suffix));
+                    }
+                }
+            }
+        }
+
+        private void updateNamedOperationResolverHook(final GraphConfig config, final List<GraphHook> hooks) {
+            if (store.isSupported(AddNamedOperation.class)) {
+                final OperationHandler addNamedOperationHandler = store.getOperationHandler(AddNamedOperation.class);
+                final String suffix = addNamedOperationHandler.getClass().isAssignableFrom(AddNamedOperationHandler.class) ? ((AddNamedOperationHandler) addNamedOperationHandler).getSuffixCacheName() : properties.getCacheServiceNamedOperationSuffix(config.getGraphId());
+                if (!hasHook(hooks, NamedOperationResolver.class)) {
+                    LOGGER.warn(HANDLER_WAS_SUPPLIED_BUT_WITHOUT_A_ADDING_WITH_SUFFIX, AddNamedOperationHandler.class.getSimpleName(), NamedOperationResolver.class.getSimpleName(), NamedOperationResolver.class.getSimpleName(), suffix);
+                    config.getHooks().add(0, new NamedOperationResolver(config.getGraphId()));
+                } else {
+                    final NamedOperationResolver nvrHook = (NamedOperationResolver) hooks.stream().filter(gh -> NamedOperationResolver.class.isAssignableFrom(gh.getClass())).findAny().get();
+                    final String nvrSuffix = nvrHook.getSuffixCacheName();
+                    if (!suffix.equals(nvrSuffix)) {
+                        throw new GafferRuntimeException(String.format(HOOK_SUFFIX_ERROR_FORMAT_MESSAGE, NamedOperationResolver.class.getSimpleName(), nvrSuffix, AddNamedOperationHandler.class.getSimpleName(), suffix));
```

Comment:
Do we have a code style policy for java line wrapping? Some of these lines are very long and could be broken up e.g. something like:
```java
throw new GafferRuntimeException(
    String.format(
        HOOK_SUFFIX_ERROR_FORMAT_MESSAGE,
        NamedOperationResolver.class.getSimpleName(),
        nvrSuffix,
        AddNamedOperationHandler.class.getSimpleName(),
        suffix
    )
);
```

Reply:
We should be following the Google Java style guide ([as stated on ways of working in docs](https://gchq.github.io/gaffer-doc/latest/dev/ways-of-working/#coding-style)). [This has a limit of 100 chars](https://google.github.io/styleguide/javaguide.html#s4.4-column-limit), so this should be wrapped to some extent. I've not checked if that suggestion is in line with the guide though.

The spotless plugin we are already using can enforce this, might be worth considering enabling this.

Reply:
So that how I have the format in the tests because there is 6 variants and values in the format message vary and matter for the tests.

Here all you need to know is it's an error message for `HOOK_SUFFIX_ERROR_FORMAT_MESSAGE` the rest of it is visual noise. The next 50 times you come to this class and this builder you will not be interested in the contents of the error message, only that a) it was thrown here b) its about suffix is wrong.

for visual noise in code and where it is of low value, I allow it run off into the horizon because it out the way and far far easier to scan the rest of the code with the human eye.

Reply:
similar to line 889 `nvrHook` all you care about it I've got the hook, the details of how I got it within a stream is just visual noise. UNLESS in the future wrong hook is being returned, then that line will be useful.

but the MEAT of this method is a) get the hook b) get the hookSuffix  c) does the two suffix match.